docs: overview agents (#4280)
langchain-ai/langgraph: docs/docs/agents/overview.md

---
title: Overview
---

# Agent development with LangGraph

> **LangGraph** provides both low-level primitives and high-level prebuilt components for building agent-based applications. This page focuses on the **prebuilt**, **reusable** components designed to help you construct agentic systems quickly and reliably—without the need to implement orchestration, memory, or human feedback handling from scratch.

Reviewer: wonder if there is a better word than "page" - "This section focuses..."?

> **LangGraph** provides both low-level primitives and high-level prebuilt components for building agent-based applications. This page focuses on the **prebuilt**, **reusable** components designed to help you construct agentic systems quickly and reliably—without the need to implement orchestration, memory, or human feedback handling from scratch.

Author: ```suggestion
**LangGraph** provides both low-level primitives and high-level prebuilt components for building agent-based applications. This section focuses on the **prebuilt**, **reusable** components designed to help you construct agentic systems quickly and reliably—without the need to implement orchestration, memory, or human feedback handling from scratch.
```

## Key features

LangGraph includes several capabilities essential for building robust, production-ready agentic systems:

- [**Memory integration**](./memory.md): Native support for *short-term* (session-based) and *long-term* (persistent across sessions) memory, enabling stateful behaviors in chatbots and assistants.
- [**Human-in-the-loop control**](./human-in-the-loop.md): Execution can pause *indefinitely* to await human feedback—unlike websocket-based solutions limited to real-time interaction. This enables asynchronous approval, correction, or intervention at any point in the workflow.
- [**Streaming support**](./streaming.md): Real-time streaming of agent state, model tokens, tool outputs, or combined streams.

> - [**Deployment tooling**](./deployment.md): Includes infrastructure-free deployment tools. The [**LangGraph Platform**](https://langchain-ai.github.io/langgraph/concepts/langgraph_platform/) supports testing, debugging, and deployment.

Reviewer: ditch "The" - just LangGraph Platform

> - [**Deployment tooling**](./deployment.md): Includes infrastructure-free deployment tools. The [**LangGraph Platform**](https://langchain-ai.github.io/langgraph/concepts/langgraph_platform/) supports testing, debugging, and deployment.

Author: ```suggestion
- [**Deployment tooling**](./deployment.md): Includes infrastructure-free deployment tools. [**LangGraph Platform**](https://langchain-ai.github.io/langgraph/concepts/langgraph_platform/) supports testing, debugging, and deployment.
```

## High-level building blocks

LangGraph comes with a set of prebuilt components that implement common agent behaviors and workflows. These abstractions are built on top of the LangGraph framework, offering a faster path to production while remaining flexible for advanced customization.

Using LangGraph for agent development allows you to focus on your application's logic and behavior, instead of building and maintaining the supporting infrastructure for state, memory, and human feedback.

## Package ecosystem

| Package                                    | Description                                                                 | Installation                            |
|--------------------------------------------|-----------------------------------------------------------------------------|-----------------------------------------|
| `langgraph-prebuilt` (part of `langgraph`) | Prebuilt components to [**create agents**](./agents.md)                     | `pip install -U langgraph langchain`    |
| `langgraph-supervisor`                     | Tools for building [**supervisor**](./multi-agent.md#supervisor) agents     | `pip install -U langgraph-supervisor`   |

> | `langgraph-swarm`                          | Tools for building [**swarm**](./multi-agent.md#swarm) agents               | `pip install -U langgraph-swarm`        |

Author: ```suggestion
| `langgraph-swarm`                          | Tools for building a [**swarm**](./multi-agent.md#swarm) multi-agent system  | `pip install -U langgraph-swarm`        |
```

> | `langgraph-swarm`                          | Tools for building [**swarm**](./multi-agent.md#swarm) agents               | `pip install -U langgraph-swarm`        |

Reviewer: "swarm agents" is a bit strange - those are not agents, but multi-agent systems

Reviewer: same for supervisor

Reviewer: supervisor might be fine cause it is an agent. i would just say multi-agent system?
